src/wms/src/main/java/org/geoserver/wms/GetMap.java

```diff
@@ -669,7 +669,24 @@ WebMap executeInternal(
                     }
                 }
                 if (!merged) {
-                    WMSLayer Layer = new WMSLayer(wms, gt2Layer);
+                    WMSLayer Layer = null;
+                    // GEOS-9312
+                    if (mapLayerInfo.getLayerInfo() instanceof CascadedLayerInfo) {
+                        CascadedLayerInfo cascadedLayerInfo =
+                                (CascadedLayerInfo) mapLayerInfo.getLayerInfo();
+
+                        String style = request.getStyles().get(i).getName();
+                        String imageFormat = request.getFormat();
+                        // if passed style does not exist in remote, default to forced
+                        if (!cascadedLayerInfo.isSelectedRemoteStyles(style))
+                            style = cascadedLayerInfo.getForcedRemoteStyle();
+
+                        // if the format is not selected then fall back to preffered
+                        if (!cascadedLayerInfo.isSelectedRemoteFormat(imageFormat))
+                            imageFormat = cascadedLayerInfo.getPrefferedFormat();
+
+                        Layer = new WMSLayer(wms, gt2Layer, style, imageFormat);
+                    } else Layer = new WMSLayer(wms, gt2Layer);
```

Comment:
Not sure what's the propose of this, shoudl not WMS layers always have a preferred image format and style?

Reply:
this is to retrain backward compatbility, but it will not be required after migration is in place

Reply:
kk